platform/default/src/mbgl/storage/file_source_manager.cpp

```diff
@@ -12,27 +12,28 @@ namespace mbgl {
 class DefaultFileSourceManagerImpl final : public FileSourceManager {
 public:
     DefaultFileSourceManagerImpl() {
-        registerFileSourceFactory(FileSourceType::ResourceLoader, [](const ResourceOptions& options) {
-            return std::make_unique<MainResourceLoader>(options);
+        registerFileSourceFactory(FileSourceType::ResourceLoader, [](const ResourceOptions& resourceOptions, const ClientOptions& clientOptions) {
+            return std::make_unique<MainResourceLoader>(resourceOptions, clientOptions);
         });
```

Comment:
Side question: what is platform/default used for?

Reply:
These are default implementations that may work in many other platforms so one only needs to re-implement what is not working from default. In C++ you always need a default implementation.

Reply:
Cool thanks